src/lightning_app/core/app.py

```diff
@@ -526,14 +526,11 @@ def _update_layout(self) -> None:
             component._layout = layout
 
     def _update_is_headless(self) -> None:
-        is_headless = _is_headless(self)
+        self.is_headless = _is_headless(self)
 
         # If `is_headless` changed, handle it.
         # This ensures support for apps which dynamically add a UI at runtime.
-        if self.is_headless != is_headless:
-            self.is_headless = is_headless
-
-            _handle_is_headless(self)
+        _handle_is_headless(self)
```

Comment:
why did we remove the guard here, without that wouldn't we making non necessary requests to the be?

Reply:
No, I had two guards by accident - we already have something in place that says we won't make a request unless the value changes

Reply:
cool